maven-compat/src/main/java/org/apache/maven/repository/legacy/DefaultWagonManager.java

```diff
@@ -465,6 +469,28 @@ public void getRemoteFile(
         }
     }
 
+    private void copyFile(File source, File destination) throws IOException {
```

Comment:
These sorts of methods are very tricky with lots of inobvious, platform dependent edge cases that are hard to get right. I'd be much more comfortable if we could use a single method from commons-io  like FileUtils.copyFile

Reply:
Since this is the only dependency, @gnodet suggests inlining the method,

Reply:
I wouldn't assume the existing method is correct. Please use the JDK Files.copy method instead

Reply:
It already uses `Files.copy`, so I think that's fine.  If needed, it can be inlined, but I don't see the benefit.